src/core/Form/TextInput/TextInput.tsx

```diff
@@ -86,17 +88,14 @@ export interface TextInputProps
 }
 
 class BaseTextInput extends Component<TextInputProps> {
-  private id: string;
-
   private hintTextId: string;
 
   private statusTextId: string;
 
   constructor(props: TextInputProps) {
     super(props);
-    this.id = `${idGenerator(props.id)}`;
-    this.hintTextId = `${this.id}-hintText`;
-    this.statusTextId = `${this.id}-statusText`;
+    this.hintTextId = `${props.id}-hintText`;
+    this.statusTextId = `${props.id}-statusText`;
```

Comment:
these can now be handled in render since id is handled outside the component. No need for constructor at all.